### ISSUE TITLE
High resource usage after leaving open in background for a while

### ISSUE THREAD

**Issue by Senzorei** [jixxed/ed-odyssey-materials-helper#60]
I get high CPU usage (upwards of 20%, sometimes enough to cause the whole system to lag) when leaving the program open in the background, not sure what the issue is exactly. EDIT: I've noticed it might be related to alt+tabbing somehow.

CPU - AMD Ryzen R5 2600
Memory - G.Skill 16 GB 3200 Mbit/s (F4-3200C16D-16GVKB, Hynix CJR, 16-18-18-38)
Graphics - Nvidia GTX 1050 Ti desktop (driver ver. 496.49 DCH)
Disk with log files (game and OS are each on separate drives) - WD Blue 1TB (WD10EZEX-08WN4A)

**Comment by jixxed** [comment]
What is the memory usage of the app and the system like?
Windows can become slow when you hit 80-90% memory usage.

**Comment by Senzorei** [comment]
CPU usage for Materials Helper averaging above 70%, memory at around 1.3 GiB, 75% (12 GiB or so) total system memory usage. Total size of all the Elite journal logs is 470 MiB.

This happened roughly 120-130 minutes after launch, I had EDMC running in parallel, along with a few Google Chrome tabs open (one with a Twitch stream) and a few programs with overlays (Steam, Discord, RivaTuner Statistics Server). 

I also created a memory dump of the process if you want to take a look at it, although I'd likely have to upload it elsewhere due to the size.
![EDOMH Resources](https://user-images.githubusercontent.com/85295032/146522962-2c4ab819-bd04-44b3-ae12-49303424d5db.png)

**Comment by jixxed** [comment]
a memory dump can be useful, you can try wetransfer and send me the link.

CPU usage is likely due to memory garbage collection. The memory has a top limit around 1GB + some extra in other memory spaces. memory will fill up slowly over time and when it hits a certain point it will start freeing the memory. 
This normally creates a short spike while freeing, but if memory can't get back under the limit it can keep trying infinitely, which seems to be the case. Usually this means there is some kind of memory leak, where I do keep a close eye on.

As a temporary workaround: restarting the app will set it back to normal levels. you can also increase the limit of 1GB(-Xmx1024M) in the files:
\app\Elite Dangerous Odyssey Materials Helper.cfg
\runtime\bin\Elite Dangerous Odyssey Materials Helper.bat

hopefully the memory dump can help me pinpoint the exact issue.

Another issue can be disk I/O since you run the journal on a regular HDD, although I would expect the CPU spike to be limited to a single thread.

**Comment by jixxed** [comment]
also can be helpful if you could send me the journal file that caused the issue, that way I can try to replay the issue locally.
send me:
Journal log
ShipLocker.json
Backpack.json
%localappdata%\odyssey-materials-helper\pref.properties

**Comment by Senzorei** [comment]
The sequence in the journals should start around line 724, with the item pickup being line 736.
https://we.tl/t-76vjZCCzE3

**Comment by jixxed** [comment]
There is a memory leak in the wishlist materials. due to the large amount of blueprints you had active, this issue popped up relatively fast. I've fixed the issue and it will be in with the next release.
old:
![Snag_86bf654](https://user-images.githubusercontent.com/14129491/146563365-3b328c61-e325-4b59-99a4-fcd9ed6c28e6.png)
new:
![Snag_86c0e41](https://user-images.githubusercontent.com/14129491/146563384-09d6cc95-21dc-4bb6-b8fd-722461ace683.png)

Don't mind the extreme memory values in the images, what's important is that the memory goes down to around 250MB after triggering a Garbage Collection.
